### DISCUSSION TITLE
How do I persist search params when navigating back?

### DISCUSSION THREAD

**Discussion by imtrent** [remix-run/react-router#8669] [Q&A] [answered]
react-router version - 6.2.1

Let's say I am on a route with search params: `/results?sort=name&order=desc`. These search params were set using the `useSearchParams` hook.  I then click a button that navigates me to an individual result page: `/results/123`. On the individual result page, I click a back button that calls `navigate(-1)`. How do I get the search params `?sort=name&order=desc` back? Why doesn't navigate() take search params into account when going back?

**Comment by imtrent** [comment, marked as answer]
Issue was with my own code 😅😂

**Reply by KostiantynO** [reply]
Hello, imtrent! 
Did you encounter useSearchParams + useEffect breaking history.push back functionality problem?
  Please, share the core idea behind your solution :)
How did you get it to work right? Did you just removed useEffect?
